### PR TITLE
[Text Translation] Model property names changes

### PR DESCRIPTION
**Pull request by MikeyMCZ**
* Updating names according to review
* Changing Language Directionality to enum

### CODE REVIEW

File: specification/translation/Azure.AI.TextTranslation/models-dictionary.tsp [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import "@typespec/rest";
 import "@azure-tools/typespec-azure-core";
 import "./models-shared.tsp";
+import "@azure-tools/typespec-client-generator-core";
```
Comment (joheredi) [inline specification/translation/Azure.AI.TextTranslation/models-dictionary.tsp:4]: We try to keep any customizations that need `"@azure-tools/typespec-client-generator-core"` in a separate `client.tsp` file to avoid coupling the Service definition with the Client representation. For example in the new `client.tsp` you could have

```ts
import "./models-dictionary.tsp"
import "@azure-tools/typespec-client-generator-core";

@@clientName(TextTranslation.BackTranslation.numExamples, "ExamplesCount");

@@clientName(TextTranslation.DictionaryLookupParameters.from, "SourceLanguage");

@@clientName(TextTranslation.DictionaryLookupParameters.to, "TargetLanguage");

@@clientName(TextTranslation.DictionaryExamplesParameters.from,
  "SourceLanguage"
);

@@clientName(TextTranslation.DictionaryExamplesParameters.to, "TargetLanguage");

```

File: specification/translation/Azure.AI.TextTranslation/models-languages.tsp [hunk 2]
```diff
@@ -1,15 +1,17 @@
 import "@typespec/rest";
 import "@azure-tools/typespec-azure-core";
 import "./models-shared.tsp";
+import "@azure-tools/typespec-client-generator-core";
```
Comment (joheredi) [inline specification/translation/Azure.AI.TextTranslation/models-languages.tsp:4]: Same comment as above for all the places where this import is used. You could move all of them to the same client.tsp file.

File: specification/translation/Azure.AI.TextTranslation/models-languages.tsp [hunk 3]
```diff
@@ -112,28 +114,42 @@ model CommonScriptModel {
...
 
   @doc("Language code identifying the target language.")
   code: string;
 }
+
+@doc("Language Directionality")
+union LanguageDirectionality {
+  LeftToRight: "ltr",
+  RightToLeft: "rtl",
+  string,
```
Comment (joheredi) [inline specification/translation/Azure.AI.TextTranslation/models-languages.tsp:146]: Do we really need to make this directionality open-ended? Do we expect customers to pass a custom directionality that is not defined here?

Comment (joheredi) [reply specification/translation/Azure.AI.TextTranslation/models-languages.tsp:146]: We should consider the same for the unions `ProfanityAction` and `ProfanityMarker`, really for all Unions, make sure we really want to make the set open ended

Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/models-languages.tsp]: Done for all fixed enums.

File: specification/translation/Azure.AI.TextTranslation/models-languages.tsp (hunk 2 above)
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/models-languages.tsp]: Done

File: specification/translation/Azure.AI.TextTranslation/models-dictionary.tsp (hunk 1 above)
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/models-dictionary.tsp]: Done

File: specification/translation/Azure.AI.TextTranslation/main.tsp [hunk 4]
```diff
@@ -2,6 +2,7 @@ import "@typespec/rest";
 import "@typespec/versioning";
 import "@azure-tools/typespec-azure-core";
 import "./routes.tsp";
+import "./client-names.tsp";
```
Comment (johanste) [inline specification/translation/Azure.AI.TextTranslation/main.tsp:5]: Do not import client-names.tsp in main.tsp. And rename client-names to client.tsp.

File: specification/translation/Azure.AI.TextTranslation/models-translate.tsp [hunk 5]
```diff
@@ -151,28 +151,30 @@ union TextType {
...
-  /** Replace the profanity with a specific marker. */
+  @doc("Profanity is marked.")
   Marked: "Marked",
 
-  /** Delete the profanity. */
+  @doc("Profanity is deleted from the translated text.")
   Deleted: "Deleted",
 }
 
+#suppress "@azure-tools/typespec-azure-core/use-extensible-enum" "Enums should be defined without the `@fixed` decorator."
```
Comment (johanste) [inline specification/translation/Azure.AI.TextTranslation/models-translate.tsp:169]: Why is this being suppressed?

File: specification/translation/Azure.AI.TextTranslation/routes.tsp [hunk 6]
```diff
@@ -46,9 +46,9 @@ op CustomGetOperation<TParams, TResponse, Traits = {}, TError = ErrorResponse>(
 @route("languages")
 @summary("Gets the set of languages currently supported by other operations of the Translator.")
 @doc("Gets the set of languages currently supported by other operations of the Translator.")
-op getLanguages is CustomGetOperation<
-  GetLanguageParameters,
-  GetLanguagesResult,
+op getSupportedLanguages is CustomGetOperation<
+  GetSupportedLanguagesParameters,
+  GetSupportedLanguagesResult,
```
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply specification/translation/Azure.AI.TextTranslation/routes.tsp:51]: We should probably make all of these changes in client.tsp

File: specification/translation/Azure.AI.TextTranslation/client-names.tsp
Comment (catalinaperalta) [inline specification/translation/Azure.AI.TextTranslation/client-names.tsp:1]: Here are the docs on how to customize clients and add renames: https://azure.github.io/typespec-azure/docs/howtos/DataPlane%20Generation%20-%20DPG/04renaming

The entire section under "DataPlane Generation" has information on various client configurations you can make

File: specification/translation/Azure.AI.TextTranslation/routes.tsp (hunk 6 above)
[the remark this answers is not in the document]
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/routes.tsp:51]: This is based on the GA review comment. What is the reason to have it as change in client.tsp?

File: specification/translation/Azure.AI.TextTranslation/models-translate.tsp (hunk 5 above)
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/models-translate.tsp:169]: It came after the discussion in TypeSpec channel that if we want to use enums, we need to suppress. I will tag you un the conversation.

File: specification/translation/Azure.AI.TextTranslation/main.tsp (hunk 4 above)
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/main.tsp:5]: With that change, I can see that the custom names were removed, I don't think it is expected:
image

Comment (johanste) [reply specification/translation/Azure.AI.TextTranslation/main.tsp:5]: Code generation will not use the generated openapi document. You should only use the code generation decorators to control the client library.

File: specification/translation/Azure.AI.TextTranslation/routes.tsp (hunk 6 above)
[the remark this answers is not in the document]
Comment (johanste) [reply specification/translation/Azure.AI.TextTranslation/routes.tsp:51]: @catalinaperalta, renaming these operations to match between client library and rest api documentation is overall goodness.

File: specification/translation/Azure.AI.TextTranslation/main.tsp (hunk 4 above)
Comment (MikeyMCZ) [reply specification/translation/Azure.AI.TextTranslation/main.tsp]: Changed.

**Review by johanste — APPROVED** [review]

File: specification/translation/Azure.AI.TextTranslation/models-translate.tsp [hunk 7]
```diff
@@ -144,35 +144,37 @@ model TranslationResult {
...
+  @doc("Plain text.")
   Plain: "Plain",
 
-  /** HTML-encoded text. */
+  @doc("HTML-encoded text.")
   Html: "Html",
 }
 
+#suppress "@azure-tools/typespec-azure-core/use-extensible-enum" "Enums should be defined without the `@fixed` decorator."
+#suppress "@azure-tools/typespec-azure-core/no-enum" "Azure services should not use the enum keyword."
```
Comment (catalinaperalta) [inline specification/translation/Azure.AI.TextTranslation/models-translate.tsp:155]: Can you open an issue for yourself to update these to more meaningful suppression comments? These look like the errors that are reported instead of the reason why your service needs to suppress these warnings. Can be in a separate PR
